plugins/git-plugin/src/main/groovy/org/rundeck/plugin/scm/git/GitExportPlugin.groovy

```diff
@@ -118,7 +120,21 @@ class GitExportPlugin extends BaseGitPlugin implements ScmExportPlugin {
         File base = new File(config.dir)
         mapper = new TemplateJobFileMapper(expand(config.pathTemplate, [format: config.format], "config"), base)
         cloneOrCreate(context, base, config.url)
-
+        //check clone was ok
+        if (git?.repository.getFullBranch() != "refs/heads/$branch") {
+            logger.debug("branch differs")
+            if(config.shouldCreateBranch()){
+                if(config.baseBranch && existBranch("refs/remotes/${this.REMOTE_NAME}/${config.baseBranch}")){
+                    createBranch(context, config.branch, config.baseBranch)
+                    cloneOrCreate(context, base, config.url)
+                }else{
+                    logger.debug("Non existent remote branch: ${config.baseBranch}")
+                    throw new ScmPluginException("Non existent remote branch: ${config.baseBranch}")
+                }
+            }else{
+                throw new ScmPluginException("Non existent remote branch: ${this.branch}")
```

Comment:
Maybe the error message in this case should say something like `Could not clone the remote branch: ${this.branch}, because it does not exist. To create it, you need to set the Create Branch option to true.` So that the user knows how to fix the problem.